kyo-scheduler/jvm/src/main/scala/kyo/scheduler/Scheduler.scala

```diff
@@ -176,22 +178,25 @@ final class Scheduler(
         }
 
     private val cycleTask =
-        timerExecutor.scheduleAtFixedRate(
-            () => cycleWorkers(),
-            cycleNs,
-            cycleNs,
-            TimeUnit.NANOSECONDS
+        timerExecutor.submit(
+            (
+                () => {
+                    val thread = Thread.currentThread()
+                    while (!thread.isInterrupted()) {
+                        cycleWorkers()
+                        LockSupport.parkNanos(cycleIntervalNs)
+                    }
+                }
+            ): Callable[Unit]
         )
 
     private def cycleWorkers(): Unit = {
         try {
             val nowMs    = clock.currentMillis()
             var position = 0
-            while (position < allocatedWorkers) {
+            while (position < currentWorkers) {
```

Comment:
There was no need to cycle idle workers since they can't receive new workload and exit as soon as possible.